src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -3802,6 +3818,64 @@ SubqueryPushdownMultiNodeTree(Query *queryTree)
 }
 
 
+/*
+ * FlattenJoinVars iterates over provided columnList to identify
```

Comment:
I think the comment is pretty clear. 

Maybe one thing to mention is that Postgres handles this during standard planner and we don't use the modified query? (i.e., just before or after the sentence `Postgres is smart about it to detect that ..`)

Reply:
done